sessionctx/binloginfo/binloginfo.go

```diff
@@ -154,11 +155,8 @@ func addSpecialComment(ddlQuery string) string {
 	if strings.Contains(ddlQuery, specialPrefix) {
 		return ddlQuery
 	}
-	upperQuery := strings.ToUpper(ddlQuery)
-	reg, err := regexp.Compile(`SHARD_ROW_ID_BITS\s*=\s*\d+`)
-	terror.Log(err)
-	loc := reg.FindStringIndex(upperQuery)
-	if len(loc) < 2 {
+	loc := shardPat.FindStringIndex(strings.ToUpper(ddlQuery))
+	if loc == nil {
```

Comment:
why change the logic?

Reply:
This is the documented way to check for "no match":

> FindStringIndex returns a two-element slice of integers defining the location of the leftmost match in s of the regular expression. The match itself is at s[loc[0]:loc[1]]. A return value of nil indicates no match.